etf-sdk/src/lib.rs

```diff
@@ -1,237 +1,271 @@
+    let secret_key = SecretKey::<TinyBLS377>(msk);
+    // Look into String Formatting
+    let pp_conversion: Vec<u8> = serde_wasm_bindgen::from_value(p_pub_js.clone())
+        .map_err(|_| JsError::new("could not decode p_pub"))?;
+    let pp_bytes: [u8;144] = pp_conversion.try_into().map_err(|_| JsError::new("could not convert public params"))?;
+    let double_pub_key = convert_from_bytes::<DoublePublicKey<TinyBLS377>, 144>(&pp_bytes.clone()).ok_or(JsError::new("Could not convert secret key"))?;
+    let pp = double_pub_key.1;
+    let id_bytes: Vec<u8> = serde_wasm_bindgen::from_value(id_js.clone())
+        .map_err(|_| JsError::new("could not decode id"))?;
+    let identity = Identity::new(&id_bytes);
+    let message_bytes: Vec<u8> = serde_wasm_bindgen::from_value(message_js.clone())
+        .map_err(|_| JsError::new("could not decode message"))?;
+    let mut ciphertext_bytes: Vec<_> = Vec::new();
+    let ciphertext = secret_key.encrypt(pp, &message_bytes, identity, &mut rng).map_err(|_| JsError::new("encryption has failed"))?;
+    ciphertext.serialize_compressed(&mut ciphertext_bytes).map_err(|_| JsError::new("ciphertext serialization has failed"))?;
+    
+    serde_wasm_bindgen::to_value(&ciphertext_bytes).map_err(|_| JsError::new("could not convert ciphertext to JsValue"))
+}
+
+/// The decrypt wrapper used by the WASM blob to call tlock.rs encrypt function in etf-crypto-primitives
+/// * 'ciphertext_js': The string to be decrypted
+/// * 'sig_vec_js': The array of BLS signatures required to rebuild the secret key and decrypt the message
+#[wasm_bindgen]
+pub fn decrypt(
+    ciphertext_js: JsValue,
+    sig_vec_js: JsValue
+) -> Result<JsValue, JsError>{
+
+    let sig_conversion: Vec<u8> = serde_wasm_bindgen::from_value(sig_vec_js.clone())
+         .map_err(|_| JsError::new("could not decode secret key"))?;
+
+    let sig_bytes = sig_conversion.as_slice();
+    let sig_vec: Vec<IBESecret<TinyBLS377>> = Vec::deserialize_compressed(sig_bytes).map_err(|_| JsError::new("could not deserialize sig_vec"))?;
+    
+    let ciphertext_vec: Vec<u8> = serde_wasm_bindgen::from_value(ciphertext_js.clone())
+        .map_err(|_| JsError::new("could not decode ciphertext"))?;
+    let ciphertext_bytes: &[u8] = ciphertext_vec.as_slice();
+
+    let ciphertext:TLECiphertext<TinyBLS377> = TLECiphertext::deserialize_compressed(ciphertext_bytes).map_err(|_| JsError::new("Could not deserialize ciphertext"))?;
+    let decrypt_result: DecryptionResult= ciphertext.decrypt(sig_vec).map_err(|_| JsError::new("decryption has failed"))?;
+    let message: Vec<u8> = decrypt_result.message;
+    let plaintext: String = String::from_utf8(message).map_err(|_| JsError::new("Plaintext could not be converted to a string"))?;
+
+    serde_wasm_bindgen::to_value(&plaintext).map_err(|_| JsError::new("plaintext conversion has failed"))
+}
+
+/// Temporary logging struct
```

Comment:
still needed?

Reply:
I'd say it's not necessary, but it is highly convenient for working on the file.